### PR TITLE
fix(vercel): Switch to node 18 when local version is not supported

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes

- #7717 
- Switch to Node.js 18 runtime when the version being used locally isn't supported by Vercel Serverless Functions.
- Warn when the local version being used is deprecated by Vercel.

#### What users will see when using an unsupported version
image

#### What users will see when using a deprecated version
image

## Docs
- Should not be required as this does not affect usage.

### CODE REVIEW

**Review by natemoo-re — APPROVED** [review]
This looks great, thank you!

File: packages/integrations/vercel/src/serverless/adapter.ts [hunk 1]
```diff
@@ -19,6 +19,12 @@ const PACKAGE_NAME = '@astrojs/vercel/serverless';
 export const ASTRO_LOCALS_HEADER = 'x-astro-locals';
 export const VERCEL_EDGE_MIDDLEWARE_FILE = 'vercel-edge-middleware';
 
+const SUPPORTED_NODE_VERSIONS: Record<string, { status: 'current' } | { status: 'deprecated', removal: string }> = {
```
Comment (bluwy) [inline packages/integrations/vercel/src/serverless/adapter.ts:22]: Would be nice to add a comment and link to https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version so we can keep this up-to-date in the future.

File: packages/integrations/vercel/src/serverless/adapter.ts [hunk 2]
```diff
@@ -19,6 +19,12 @@ const PACKAGE_NAME = '@astrojs/vercel/serverless';
...
+	14: { status: 'deprecated', removal: 'August 15th 2023'  },
+	16: { status: 'deprecated', removal: 'February 6th 2024' },
```
Comment (bluwy) [inline packages/integrations/vercel/src/serverless/adapter.ts:24]: Ah the US date format 😅

**Review by bluwy — APPROVED** [review]

File: packages/integrations/vercel/src/serverless/adapter.ts (hunk 1 above)
Comment (lilnasy) [reply packages/integrations/vercel/src/serverless/adapter.ts]: Yes, that's a great idea. Meant to do that actually.

File: packages/integrations/vercel/src/serverless/adapter.ts (hunk 2 above)
Comment (lilnasy) [reply packages/integrations/vercel/src/serverless/adapter.ts]: Using Intl.DateTimeFormat now 🫡

Comment (bluwy) [reply packages/integrations/vercel/src/serverless/adapter.ts]: Meant that mostly as a joke, but now it's even better! Thanks 🙌

**Review by bluwy — APPROVED** [review]

**Review by ematipico — APPROVED** [review]

**Pull request merged**